packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -70,14 +72,59 @@ export function Comments( {
 		setIsConfirmDialogOpen( false );
 	};
 
-	const blockCommentId = useSelect( ( select ) => {
+	useSelect( ( select ) => {
 		const clientID = select( blockEditorStore ).getSelectedBlockClientId();
-		return (
+		const clientBlocks = select( blockEditorStore ).getBlocks();
+		setBlocksList( clientBlocks );
+
+		const getBlockCommentId =
 			select( blockEditorStore ).getBlock( clientID )?.attributes
-				?.blockCommentId ?? false
-		);
+				?.blockCommentId ?? false;
+
+		if ( getBlockCommentId ) {
+			setActiveClientId( getBlockCommentId );
+		}
 	}, [] );
 
+	const findBlockByCommentId = ( blocks, commentId ) => {
+		for ( const block of blocks ) {
+			if ( block.attributes.blockCommentId === commentId ) {
+				return block;
+			}
+			if ( block.innerBlocks && block.innerBlocks.length > 0 ) {
+				const foundBlock = findBlockByCommentId(
+					block.innerBlocks,
+					commentId
+				);
+				if ( foundBlock ) {
+					return foundBlock;
+				}
+			}
+		}
+		return null;
+	};
+
+	const handleThreadClick = ( thread ) => {
+		const block = findBlockByCommentId( blocksList, thread.id );
+
+		if ( block ) {
+			const blockClientId = block.clientId;
+			const iframe = document.querySelector(
+				'iframe[name="editor-canvas"]'
```

Comment:
Let's use `useBlockElementRef`

Reply:
@ellatrix - where I try to use the useBlockElementRef hook, As per my understanding, it’s a part of the @wordpress/block-editor package. However, when I attempt to use it in the @wordpress/editor package, I cannot get it to work as expected.

Could you please confirm if this is an expected limitation? If so, are there any recommended workarounds or alternative approaches to achieve similar functionality (e.g., getting a reference to a block’s DOM element within the editor context)?

Thanks in advance.

Reply:
You mean that it's not available as an API? Can we export it as a private API so we can use it?

Reply:
If just need to focus the respective block when comment board is focus, I think below code would do the work

`dispatch( 'core/block-editor' ).selectBlock( block?.clientId );`